### PR TITLE
Feat/tanlang/user data isolation / 用户数据隔离

### PR DESCRIPTION
**Pull request by LinZexiao**
## 关联的Issues (Related Issues)

involve https://github.com/filecoin-project/venus/issues/5384

## 自查清单 (Checklist)

在你认为本 PR 满足被审阅的标准之前，需要确保 / Before you mark the PR ready for review, please make sure that:
- [x] 符合Venus项目管理规范中关于PR的[相关标准](https://github.com/ipfs-force-community/dev-guidances/blob/master/%E9%A1%B9%E7%9B%AE%E7%AE%A1%E7%90%86/Venus/PR%E5%91%BD%E5%90%8D%E8%A7%84%E8%8C%83.md) / The PR follows the PR standards set out in the Venus project management guidelines
- [x] 具有清晰明确的[commit message](https://github.com/ipfs-force-community/dev-guidances/blob/master/%E8%B4%A8%E9%87%8F%E7%AE%A1%E7%90%86/%E4%BB%A3%E7%A0%81/git%E4%BD%BF%E7%94%A8/commit-message%E9%A3%8E%E6%A0%BC%E8%A7%84%E8%8C%83.md) / All commits have a clear commit message.
- [x] 包含相关的的[测试用例](https://github.com/ipfs-force-community/dev-guidances/blob/master/%E8%B4%A8%E9%87%8F%E7%AE%A1%E7%90%86/%E4%BB%A3%E7%A0%81/%E4%BB%A3%E7%A0%81%E5%BA%93/%E6%A3%80%E6%9F%A5%E9%A1%B9/%E5%8D%95%E5%85%83%E6%B5%8B%E8%AF%95.md)或者不需要新增测试用例 / This PR has tests for new functionality or change in behaviour or not need to add new tests.
- [x] 包含相关的的指南以及[文档](https://github.com/ipfs-force-community/dev-guidances/tree/master/%E8%B4%A8%E9%87%8F%E7%AE%A1%E7%90%86/%E6%96%87%E6%A1%A3)或者不需要新增文档 / This PR has updated usage guidelines and documentation or not need 
- [ ] 通过必要的检查项 / All checks are green

### 不需要做用户数据隔离的接口
PiecesListPieces   //perm:read
PiecesListCidInfos //perm:read
PiecesGetPieceInfo //perm:read
PiecesGetCIDInfo   //perm:read
MessagerWaitMessage //perm:read
MessagerGetMessage  //perm:read
NetAddrsListen //perm:read
ID             //perm:read
ResponseMarketEvent //perm:read
ListenMarketEvent   //perm:read
ListPieceStorageInfos  //perm:read
MarketListAsk          //perm:read
MarketListRetrievalAsk //perm:read

### 不对普通用户开放的接口
MarketListDataTransfers   //perm:admin
MarketDataTransferUpdates //perm:admin
MarketRestartDataTransfer //perm:admin
MarketCancelDataTransfer  //perm:admin
ImportV1Data              //perm:admin
MarketPublishPendingDeals //perm:admin
DagstoreListShards        //perm:admin
DagstoreInitializeShard   //perm:admin
DagstoreRecoverShard      //perm:admin
DagstoreInitializeAll     //perm:admin
DagstoreInitializeStorage //perm:admin
DagstoreGC                //perm:admin
AddFsPieceStorage         //perm:admin
AddS3PieceStorage         //perm:admin
RemovePieceStorage        //perm:admin

### 暂时不进行用户数据隔离的接口
MarketListRetrievalDeals //perm:read

### 进行用户数据隔离的接口
ActorList       //perm:read
ActorExist      //perm:read
ActorSectorSize //perm:read
MarketListDeals //perm:read

MarketGetDealUpdates      //perm:read
MarketListIncompleteDeals //perm:read
MarketGetAsk              //perm:read

MarketGetRetrievalAsk //perm:read

DealsConsiderOnlineStorageDeals     //perm:read
DealsConsiderOnlineRetrievalDeals   //perm:read
DealsPieceCidBlocklist              //perm:read
DealsConsiderOfflineStorageDeals    //perm:read
DealsConsiderOfflineRetrievalDeals  //perm:read
DealsConsiderVerifiedStorageDeals   //perm:read
DealsConsiderUnverifiedStorageDeals //perm:read
SectorGetExpectedSealDuration       //perm:read
DealsMaxStartDelay                  //perm:read

DealsPublishMsgPeriod       //perm:read
MarketMaxDealsPerPublishMsg //perm:read

DealsMaxProviderCollateralMultiplier //perm:read
DealsMaxPublishFee                   //perm:read
MarketMaxBalanceAddFee               //perm:read

GetDeals         //perm:read
GetUnPackedDeals //perm:read

PaychVoucherList //perm:read

GetStorageDealStatistic   //perm:read
GetRetrievalDealStatistic //perm:read

MarketImportDealData      //perm:write
MarketImportPublishedDeal //perm:write

MarketPendingDeals                     //perm:write
DealsSetConsiderOnlineStorageDeals     //perm:write
DealsSetConsiderOnlineRetrievalDeals   //perm:write
DealsSetPieceCidBlocklist              //perm:write
DealsSetConsiderOfflineStorageDeals    //perm:write
DealsSetConsiderOfflineRetrievalDeals  //perm:write
DealsSetConsiderVerifiedStorageDeals   //perm:write
DealsSetConsiderUnverifiedStorageDeals //perm:write
SectorSetExpectedSealDuration          //perm:write
DealsSetMaxStartDelay                  //perm:write

DealsSetPublishMsgPeriod                //perm:write
MarketSetMaxDealsPerPublishMsg          //perm:write
DealsSetMaxProviderCollateralMultiplier //perm:write
DealsSetMaxPublishFee                   //perm:write
MarketSetMaxBalanceAddFee               //perm:write
MessagerPushMessage                     //perm:write
MarkDealsAsPacking                      //perm:write
UpdateDealOnPacking                     //perm:write
UpdateDealStatus                        //perm:write
AssignUnPackedDeals                     //perm:write
UpdateStorageDealStatus                 //perm:write

MarketAddBalance   //perm:sign
MarketGetReserved  //perm:sign
MarketReserveFunds //perm:sign
MarketReleaseFunds //perm:sign
MarketWithdraw     //perm:sign

MarketSetAsk              //perm:admin
MarketSetRetrievalAsk     //perm:admin
DealsImportData           //perm:admin
OfflineDealImport         //perm:admin
MarketDataTransferPath    //perm:admin
MarketSetDataTransferPath //perm:admin

#### 删除
ImportV1Data              //perm:admin

#### 权限升级
- read -> admin
MarketListDataTransfers   //perm:admin
MarketDataTransferUpdates //perm:admin
MarketRestartDataTransfer //perm:admin
MarketCancelDataTransfer  //perm:admin

### CODE REVIEW

File: api/impl/venus_market.go [hunk 1]
```diff
@@ -235,14 +291,24 @@ func (m *MarketNodeImpl) MarketSetAsk(ctx context.Context, mAddr address.Address
 }
 
 func (m *MarketNodeImpl) MarketListAsk(ctx context.Context) ([]*types.SignedStorageAsk, error) {
-	return m.StorageAsk.ListAsk(ctx)
+	asks, err := m.StorageAsk.ListAsk(ctx)
```
Comment (simlecode) [inline api/impl/venus_market.go:294]: 这里不校验？

File: api/impl/venus_market.go [hunk 2]
```diff
@@ -319,7 +388,14 @@ func (m *MarketNodeImpl) MarketCancelDataTransfer(ctx context.Context, transferI
 }
 
 func (m *MarketNodeImpl) MarketPendingDeals(ctx context.Context) ([]types.PendingDealInfo, error) {
-	return m.DealPublisher.PendingDeals(), nil
+	dealInfos := m.DealPublisher.PendingDeals()
+	ret := make([]types.PendingDealInfo, len(dealInfos))
```
Comment (simlecode) [inline api/impl/venus_market.go:392]: 你那样写前面几个都是空的，改成 `ret := make([]types.PendingDealInfo, 0, len(dealInfos))`

File: api/impl/venus_market.go [hunk 3]
```diff
@@ -125,14 +134,30 @@ func (m MarketNodeImpl) ListenMarketEvent(ctx context.Context, policy *gatewayTy
...
 
 func (m MarketNodeImpl) ActorList(ctx context.Context) ([]types.User, error) {
-	return m.UserMgr.ActorList(ctx)
+	actors, err := m.UserMgr.ActorList(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ret := []types.User{}
+	for _, actor := range actors {
+		if err := m.checkPermissionByName(ctx, actor.Account); err != nil {
```
[the remark this answers is not in the document]
Comment (LinZexiao) [reply api/impl/venus_market.go:143]: 已修改

File: api/impl/venus_market.go (hunk 2 above)
Comment (LinZexiao) [reply api/impl/venus_market.go:392]: 已修改

File: api/impl/venus_market.go [hunk 4]
```diff
@@ -1074,9 +1222,115 @@ func (m *MarketNodeImpl) GetStorageDealStatistic(ctx context.Context, miner addr
...
 	return &types.RetrievalDealStatistic{DealsStatus: statistic}, nil
 }
+
+func (m *MarketNodeImpl) checkPermissionBySigner(ctx context.Context, addrs ...address.Address) error {
+	if auth.HasPerm(ctx, []auth.Permission{}, core.PermAdmin) {
+		return nil
+	}
+	user, exist := jwtclient.CtxGetName(ctx)
+	if !exist {
+		return nil
```
[the remark this answers is not in the document]
Comment (LinZexiao) [reply api/impl/venus_market.go:1241]: 已修改

File: api/impl/venus_market.go (hunk 1 above)
Comment (LinZexiao) [reply api/impl/venus_market.go:294]: 已修改

Comment (simlecode) [reply api/impl/venus_market.go:294]: 那就按原来的写法，直接返回就行

Comment (LinZexiao) [reply api/impl/venus_market.go]: ok

File: api/impl/venus_market.go [hunk 5]
```diff
@@ -146,6 +171,13 @@ func (m MarketNodeImpl) ActorSectorSize(ctx context.Context, addr address.Addres
 }
 
 func (m *MarketNodeImpl) MarketImportDealData(ctx context.Context, propCid cid.Cid, path string) error {
+	deal, err := m.Repo.StorageDealRepo().GetDeal(ctx, propCid)
```
Comment (hunjixin) [inline api/impl/venus_market.go:174]: 离线订单的导入只有market的维护人员才能操作，这是无法远程操作的， 因此这里给admin权限。

Comment (LinZexiao) [reply api/impl/venus_market.go:174]: 已经将权限提升为 admin

**Review by hunjixin — APPROVED** [review]

**Pull request merged**